components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java

```diff
@@ -2886,6 +2887,399 @@ private OMElement queryFirstAccess(String columnFamily)
+            throws APIMgtUsageQueryServiceClientException {
+
+        if (dataSource == null) {
+            throw new APIMgtUsageQueryServiceClientException("BAM data source hasn't been initialized. Ensure " +
+                    "that the data source is properly configured in the APIUsageTracker configuration.");
+        }
+
+        Connection connection = null;
+        PreparedStatement statement = null;
+        ResultSet rs = null;
+        try {
+            connection = dataSource.getConnection();
+            String query, groupByStmt;
+            List<APIThrottlingOverTimeDTO> throttlingData = new ArrayList<APIThrottlingOverTimeDTO>();
+            String tenantDomain = MultitenantUtils.getTenantDomain(provider);
+
+            //check whether table exist first
+            if (isTableExist(APIUsageStatisticsClientConstants.API_THROTTLED_OUT_SUMMARY, connection)) { //Table exists
+                
+                if (APIUsageStatisticsClientConstants.GROUP_BY_DAY.equals(groupBy)){
+                    groupByStmt = "year, month, day";
+                } else if (APIUsageStatisticsClientConstants.GROUP_BY_HOUR.equals(groupBy)){
+                    groupByStmt = "year, month, day, time";
+                } else {
+                    throw new APIMgtUsageQueryServiceClientException(
+                            "Unsupported group by parameter " + groupBy +
+                                    " for retrieving throttle data of API and app.");
+                }
+
+                query = "SELECT " + groupByStmt + " ," +
+                        "SUM(COALESCE(success_request_count,0)) AS success_request_count, " +
+                        "SUM(COALESCE(throttleout_count,0)) AS throttleout_count " +
+                        "FROM API_THROTTLED_OUT_SUMMARY " +
+                        "WHERE tenantDomain = ? " +
+                        "AND api = ? " +
+                        (provider.startsWith(APIUsageStatisticsClientConstants.ALL_PROVIDERS) ? "" :
+                                "AND apiPublisher = ?") +
+                        (StringUtils.isEmpty(appName) ? "" : " AND applicationName = ?") +
+                        "AND time BETWEEN ? AND ? " +
+                        "GROUP BY " + groupByStmt + " " +
+                        "ORDER BY " + groupByStmt + " ASC";
+
+                statement = connection.prepareStatement(query);
+                int index = 1;
+                statement.setString(index++, tenantDomain);
+                statement.setString(index++, apiName);
+                if (!provider.startsWith(APIUsageStatisticsClientConstants.ALL_PROVIDERS)) {
+                    statement.setString(index++, provider);
+                }
+                if (!StringUtils.isEmpty(appName)) {
+                    statement.setString(index++, appName);
+                }
+                statement.setString(index++, fromDate);
+                statement.setString(index, toDate);
+
```

Comment:
Is there any special reason to use this 'index' variable? If not, I suggest to avoid that to make it more readable.